apps/docs/pages/guides/auth/auth-hooks.mdx

```diff
@@ -0,0 +1,717 @@
+
+Read more about `security definer` tag [in our database guide](/docs/guides/database/functions#security-definer-vs-invoker).
+
+### Hook errors
+
+You should return an error when facing a runtime error. Runtime errors are specific to your application and arise from specific business rules rather than programmer errors.
+
+Runtime errors could happen when:
+
+- The user does not have appropriate permissions
+- The event payload received does not have required claims.
+- The user has performed an action which violates a business rule.
+
+The error is a JSON object and has the following properties:
+
+- `error` An object that contains information about the error.
+  - `http_code` A number indicating the HTTP code to be returned. If not set, the code is HTTP 500 Internal Server Error.
+  - `message` A message to be returned in the HTTP response. Required.
+
+Here's an example:
+
+```json
+{
+  "error": {
+    "http_code": 429,
+    "message": "You can only verify a factor once every 10 seconds."
+  }
+}
+```
+
+When an error is returned, the error is propagated from the hook to Supabase Auth and translated into a HTTP error which is returned to your application. Supabase Auth will only take into account the error and disregard the rest of the payload.
+
+### Timeouts
+
+Ensure that your hooks complete within **2 seconds** to avoid any errors.
+
+### Connect a Hook
+
+In the dashboard, navigate to [`Authentication > Hooks (Beta)`](/dashboard/project/_/auth/hooks) and select the appropriate PostgreSQL function from the dropdown menu.
+
+## Hook: MFA Verification Attempt
+
+You can add additional checks to the [Supabase MFA implementation](/docs/guides/auth/auth-mfa) with hooks. For example, you can:
+
+- Limit the number of verification attempts performed over a period of time.
+- Sign out users who have too many invalid verification attempts.
+- Count, rate limit, or ban sign-ins.
+
+**Inputs**
+
+Supabase Auth will supply the following fields to your hook:
+
+- `factor_id` Unique identifier for the MFA factor being verified.
+- `user_id` Unique identifier for the user.
+- `valid` Whether the verification attempt was valid. For TOTP, this means that the six digit code was correct (true) or incorrect (false).
+
+Example payload:
+
+```json
+{
+  "factor_id": "6eab6a69-7766-48bf-95d8-bd8f606894db",
+  "user_id": "3919cb6e-4215-4478-a960-6d3454326cec",
+  "valid": true
+}
+```
+
+**Outputs**
+
+Return this if your hook processed the input without errors.
+
+- `decision` A string containing the decision on whether to allow authentication to move forward. Use `reject` to deny the verification attempt and log the user out of all active sessions. Use `continue` to use the default Supabase Auth behavior.
+- `message` The message to show the user if the decision was `reject`.
+
+Example output:
+
+```json
+{
+  "decision": "reject",
+  "message": "You have exceeded maximum number of MFA attempts."
+}
+```
+
+**Examples**
+
+<Tabs
```

Comment:
For all tabs in the doc, can you check how we can have the first example be active and not hidden?